azure-iot-device/azure/iot/device/common/pipeline/pipeline_stages_base.py

```diff
@@ -786,8 +786,18 @@ def on_send_request_done(op, error):
                         self.name, op_waiting_for_response.name, request_id
                     )
                 )
-                del self.pending_responses[request_id]
-                op_waiting_for_response.complete(error=error)
+                # if there's no pending response for the given request_id, there's nothing to delete
+                if request_id not in self.pending_responses:
```

Comment:
This has since been explained to me as being an edge case that comes out of cleanup operations not affecting items that have been already sent to Paho. At this time we do not wish to change how that process works, so this will be sufficient when an explanatory NOTE is adding to this conditional block

Reply:
I would also probably reverse the conditional here for readability

Reply:
We discussed the root cause of the problem, and I added that as a comment to the issue to ensure it's properly recorded. The conditional has been reversed for readability.